### DISCUSSION TITLE
Upload Limit For Trilium

### DISCUSSION THREAD

**Discussion by ChromesDuzez** [TriliumNext/Trilium#3815] [Q&A]
Why is there a limit to the size of files I can upload into Trilium? I understand why this would be very important if you were to be paying for a hosting service but for example, I self-host my own trillium server and I'm curious if I can make the max file size for upload bigger.

**Comment by sigaloid** [comment]
There is a hard limit somewhere near 5GB due to the way Javascript handles files. There's an artificial limit placed because Trilium is not designed for large file storage, but it can be disabled if you desire with an environment variable. However, it's still true that SQLite and Trilium are not meant to hold singularly large binary blobs. It increases sync failure rates because of connectivity issues and can cause slowdowns.

**Reply by zadam** [reply]
I think the limit is dependent on the platform, but IIRC problems appeared much lower, I think even below 500 MB.

**Comment by ChromesDuzez** [comment]
cool ty Ig a cool feature would be being able to hook it up to seperate db for lfs if we want to store single large blobs

**Comment by zadam** [comment]
@sigaloid is correct. Handling large files is not "free" and has to be programmed in a specific way. Trilium makes an assumption that the content will always fit into memory, SQLite has a hard limit for BLOB which can be between 1 and 2 GBs. Some operations may be very slow etc. A lot of these issues can be worked around, but given that Trilium is not meant to be a large file storage, I don't dedicate my time towards it and limit the size to a safe value instead.
